OutOfSchool/OutOfSchool.IdentityServer/Data/Migrations/OutOfSchoolMigrations/20210828133338_ActualAddressCanBeNull.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace OutOfSchool.IdentityServer.Data.Migrations.OutOfSchoolMigrations
+{
+    public partial class ActualAddressCanBeNull : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropForeignKey(
+                name: "FK_Providers_Addresses_ActualAddressId",
+                table: "Providers");
+
+            migrationBuilder.AlterColumn<long>(
+                name: "ActualAddressId",
+                table: "Providers",
+                type: "bigint",
+                nullable: true,
+                oldClrType: typeof(long),
+                oldType: "bigint");
+
+            migrationBuilder.AddForeignKey(
+                name: "FK_Providers_Addresses_ActualAddressId",
+                table: "Providers",
+                column: "ActualAddressId",
+                principalTable: "Addresses",
+                principalColumn: "Id",
+                onDelete: ReferentialAction.Restrict);
```

Comment:
Why not `ReferentialAction.Cascade` ?

Reply:
because our foreign key now nullable.

Reply:
When Cascade is set, so if we delete Actual address we will delete Provider.
May be it's not bad to set Restrict.
But consider that if we set DeleteBehavior of EF Core like Cascade and delete Address(with tracking Provider in the navigation property) it will delete Provider anyway.

Reply:
migration recreated